miasm2/analysis/dse.py

```diff
@@ -596,13 +605,20 @@ def handle_correct_destination(self, destination, path_constraints):
             self.cur_solver.add(self.z3_trans.from_expr(cons))
 
     def handle(self, cur_addr):
+        cur_addr = canonize_to_exprloc(cur_addr, self.ir_arch.symbol_pool)
         symb_pc = self.eval_expr(self.ir_arch.IRDst)
         possibilities = possible_values(symb_pc)
         cur_path_constraint = set() # path_constraint for the concrete path
         if len(possibilities) == 1:
-            assert next(iter(possibilities)).value == cur_addr
+            dst = next(iter(possibilities)).value
+            dst = canonize_to_exprloc(dst, self.ir_arch.symbol_pool)
+            assert dst == cur_addr
         else:
             for possibility in possibilities:
+                target_value = canonize_to_exprloc(
```

Comment:
If this variable name has to change, a `target_addr` could be more suited

Reply:
done